packages/contractkit/src/wrappers/Attestations.ts

```diff
@@ -179,50 +176,50 @@ export class AttestationsWrapper extends BaseWrapper<Attestations> {
   }
 
   /**
-   * Returns an array of attestations that can be completed, along with the issuers public key
+   * Returns an array of attestations that can be completed, along with the issuers attestation service url
    * @param phoneNumber
    * @param account
    */
   async getActionableAttestations(
     phoneNumber: string,
     account: Address
   ): Promise<ActionableAttestation[]> {
-    const accounts = await this.kit.contracts.getAccounts()
     const phoneHash = PhoneNumberUtils.getPhoneHash(phoneNumber)
-    const expiryBlocks = await this.attestationExpiryBlocks()
-    const currentBlockNumber = await this.kit.web3.eth.getBlockNumber()
-
-    const issuers = await this.contract.methods.getAttestationIssuers(phoneHash, account).call()
-    const issuerState = Promise.all(
-      issuers.map((issuer) =>
-        this.contract.methods
-          .getAttestationState(phoneHash, account, issuer)
-          .call()
-          .then(parseAttestationInfo)
-      )
+
+    const result = await this.contract.methods.getCompletableAttestations(phoneHash, account).call()
+
+    const metadataURLs = parseSolidityStringArray(
+      result[2].map(toNumber),
+      (result[3] as unknown) as string
     )
 
-    // Typechain is not properly typing getDataEncryptionKey
-    const publicKeys: Promise<string[]> = Promise.all(
-      issuers.map((issuer) => accounts.getDataEncryptionKey(issuer) as any)
+    const withAttestationServiceURLs = await concurrentMap(
+      5,
+      zip3(result[0].map(toNumber), result[1], metadataURLs),
+      async ([blockNumber, issuer, metadataURL]) => {
+        try {
+          const metadata = await IdentityMetadataWrapper.fetchFromURL(metadataURL)
+          const attestationServiceURLClaim = metadata.findClaim(ClaimTypes.ATTESTATION_SERVICE_URL)
+
+          if (attestationServiceURLClaim === undefined) {
+            throw new Error(`No attestation service URL registered for ${issuer}`)
+          }
+
+          // TODO: Once we have status indicators, we should check if service is up
+          // https://github.com/celo-org/celo-monorepo/issues/1586
+          return {
+            blockNumber,
+            issuer,
+            attestationServiceURL: attestationServiceURLClaim.url,
+          }
+        } catch (error) {
+          console.error(error)
+          return null
```

Comment:
Consider allowing the error to propagate

Reply:
I was thinking about it, but it seemed to me that a validator not or incorrectly registering an attestation service url should not result in an error for a client who just wants to complete attestations. Thus, I favored returning the attestations that are completable, by intent of the function.